tests/javascript/core/spec.js

```diff
@@ -41,22 +41,34 @@ define(['jquery', 'testsRoot/core/spec-setup', 'jasmineJquery'], function ($) {
 			expect(form.removeChild).toHaveBeenCalled();
 		});
 	});
-
+	
 	describe('Core Joomla.getOptions', function () {
-		it('should return options array Joomla.getOptions("com_foobar")', function () {
-			expect(Joomla.getOptions("com_foobar")).toEqual(["my options"])
-		});
-		it('should return option string Joomla.getOptions("com_foobar2")', function () {
-			expect(Joomla.getOptions("com_foobar2")).toEqual("Alert message!")
-		});
-		it('should return option Boolean false Joomla.getOptions("com_foobar3")', function () {
-			expect(Joomla.getOptions("com_foobar3")).toEqual(false)
-		});
-		it('should return default value for not existing key Joomla.getOptions("com_foobar4", 123)', function () {
-			expect(Joomla.getOptions("com_foobar4", 123)).toEqual(123)
-		});
+		it('should return options array Joomla.getOptions("com_foobar")', function (done) {
+			setTimeout(function () {
+				expect(Joomla.getOptions("com_foobar")).toEqual(["my options"]);
+				done();
+			}, 300);
+		}, 400);
+		it('should return option string Joomla.getOptions("com_foobar2")', function (done) {
+			setTimeout(function () {
+				expect(Joomla.getOptions("com_foobar2")).toEqual("Alert message!");
+				done();
```

Comment:
Why the nested delays? These can cause unpredictable results

Reply:
Most of the time this script will fail in the Travis. That's why I added nested if for takes a long time according to the [doc](https://jasmine.github.io/2.3/introduction.html#section-Asynchronous_Support). But if it makes unpredictable results will update it :)